x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_setup_state.tsx

```diff
@@ -4,52 +4,88 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { useState, useCallback, useMemo } from 'react';
+import { useState, useCallback, useMemo, useEffect } from 'react';
 
 import { isExampleDataIndex } from '../../../../common/log_analysis';
+import { ValidationIndicesError } from '../../../../common/http_api';
+import { useTrackedPromise } from '../../../utils/use_tracked_promise';
+import { callIndexPatternsValidate } from './api/index_patterns_validate';
 
 type SetupHandler = (
   indices: string[],
   startTime: number | undefined,
   endTime: number | undefined
 ) => void;
 
+export type ValidationIndicesUIError =
+  | ValidationIndicesError
+  | { error: 'NETWORK_ERROR' }
+  | { error: 'TOO_FEW_SELECTED_INDICES' };
+
+export interface ValidatedIndex {
+  index: string;
+  validation?: ValidationIndicesError;
+  checked: boolean;
+}
+
 interface AnalysisSetupStateArguments {
   availableIndices: string[];
   cleanupAndSetupModule: SetupHandler;
   setupModule: SetupHandler;
+  timestampField: string;
 }
 
-type IndicesSelection = Record<string, boolean>;
-
-type ValidationErrors = 'TOO_FEW_SELECTED_INDICES';
-
 const fourWeeksInMs = 86400000 * 7 * 4;
 
 export const useAnalysisSetupState = ({
   availableIndices,
   cleanupAndSetupModule,
   setupModule,
+  timestampField,
 }: AnalysisSetupStateArguments) => {
   const [startTime, setStartTime] = useState<number | undefined>(Date.now() - fourWeeksInMs);
   const [endTime, setEndTime] = useState<number | undefined>(undefined);
 
-  const [selectedIndices, setSelectedIndices] = useState<IndicesSelection>(
-    availableIndices.reduce(
-      (indexMap, indexName) => ({
-        ...indexMap,
-        [indexName]: !(availableIndices.length > 1 && isExampleDataIndex(indexName)),
-      }),
-      {}
-    )
+  // Prepare the validation
+  const [validatedIndices, setValidatedIndices] = useState<ValidatedIndex[]>(
+    availableIndices.map(index => ({
+      index,
+      validation: undefined,
+      checked: false,
+    }))
+  );
+  const [validateIndicesRequest, validateIndices] = useTrackedPromise(
+    {
+      cancelPreviousOn: 'resolution',
+      createPromise: async () => {
+        return await callIndexPatternsValidate(timestampField, availableIndices);
+      },
+      onResolve: ({ data }) => {
+        setValidatedIndices(
+          availableIndices.map(index => {
+            const validation = data.errors.find(error => error.index === index);
+            return {
+              index,
+              validation,
+              checked: validation === undefined && !isExampleDataIndex(index),
+            };
+          })
+        );
+      },
+      onReject: () => {
+        setValidatedIndices([]);
+      },
+    },
+    [availableIndices, timestampField]
   );
 
+  useEffect(() => {
+    validateIndices();
+  }, [availableIndices]);
```

Comment:
I'm not sure I understand how hooks work.

`validateIndices` has `availableIndices` as a dependency. If the latter changes I understand that `validateIndices` change too. Since the dependency is transitive, I could invalidate this `useEffect` using `availableIndices`... I think?

Reply:
The closure of this first argument to `useEffect` is only updated when the references in the dependency array change. A change of `validateIndices` would not trigger that so the effect would use the outdated instance of that function, which it closed over earlier.

Reply:
https://overreacted.io/a-complete-guide-to-useeffect/#dont-lie-to-react-about-dependencies goes into a lot of detail on the topic. This is why there's a linter check for this in Kibana, which is provisionally disabled in `infra` but will be enabled soon (#49610).